src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/FlowAnnotations.cs

```diff
@@ -311,16 +311,18 @@ protected override TypeAnnotations CreateValueFromKey(TypeDesc key)
                     TypeDesc baseType = key.BaseType;
                     while (baseType != null)
                     {
-                        TypeDefinition baseTypeDef = reader.GetTypeDefinition(((EcmaType)baseType.GetTypeDefinition()).Handle);
-                        typeAnnotation |= GetMemberTypesForDynamicallyAccessedMembersAttribute(reader, baseTypeDef.GetCustomAttributes());
+                        var ecmaBaseType = (EcmaType)baseType.GetTypeDefinition();
+                        TypeDefinition baseTypeDef = ecmaBaseType.MetadataReader.GetTypeDefinition(ecmaBaseType.Handle);
```

Comment:
Nit:
I wonder if there are other places with the same problem...
Maybe we should add a set of helpers to go from EcmaSomething to (reader, custom attributes).

Reply:
Yes - we could have a bit more first class support for custom attributes because we often need it in dataflow.

The type system is based on partial classes, so we could have a partial extension to EcmaType/Field/Method/Whatever to have a common interface between them that dataflow can use. Accessing attributes is awkward right now.